commit e7b89ba9d9a3314c3f16b261e32a36412024dd17
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Fri Mar 8 11:11:46 2019 +0000

    staging: vc-sm-cma: Ensure mutex and idr are destroyed
    
    map_lock and kernelid_map are created in probe, but not released
    in release should the vcsm service not connect (eg running the
    cutdown firmware).
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c b/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c
index 8ae7810f1f9a..888040e22d32 100644
--- a/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c
+++ b/drivers/staging/vc04_services/vc-sm-cma/vc_sm.c
@@ -752,7 +752,9 @@ static int bcm2835_vc_sm_cma_remove(struct platform_device *pdev)
 
 		/* Stop the videocore shared memory service. */
 		vc_sm_cma_vchi_stop(&sm_state->sm_handle);
+	}
 
+	if (sm_state) {
 		idr_destroy(&sm_state->kernelid_map);
 
 		/* Free the memory for the state structure. */
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1380_e7b89ba9d9a3314c3f16b261e32a36412024dd17.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1380_e7b89ba9d9a3314c3f16b261e32a36412024dd17.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
